ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java

```diff
@@ -180,7 +181,21 @@ private void appendIndex(QueryEntity queryEntity, AssociationKeyMetadata associa
 			fields.put( realColumnName, true );
 		}
 		queryIndex.setFields( fields );
-		queryIndex.setName( queryEntity.getTableName() + '_' + org.hibernate.ogm.util.impl.StringHelper.join( fields.keySet(), "_" ) );
+		String indexName = queryEntity.getTableName() + '_' + org.hibernate.ogm.util.impl.StringHelper.join( fields.keySet(), "_" );
+		Class<?> tableClass = context.getTableEntityTypeMapping().get( associationKeyMetadata.getAssociatedEntityKeyMetadata().getEntityKeyMetadata().getTable() );
+		javax.persistence.Table tableAnnotation = tableClass.getAnnotation( javax.persistence.Table.class );
+		if( tableAnnotation != null && tableAnnotation.indexes().length > 0) {
+			for(javax.persistence.Index index: tableAnnotation.indexes()){
+				if(!index.name().isEmpty()) {
+					indexName = index.name();
+					break;
+				}
+			}
+		}
+		if ( indexName.getBytes().length > 255 ) {
```

Comment:
How come aren't you using indexName.length()?

Reply:
Because Ignite's limit is of bytes not char length. String.length will give num of chars in the string but isn't necessarily the same as num of bytes.
Just verified in the Ignite code base as well and realised this still has a bug, Ignite specifically uses UTF_8 https://github.com/apache/ignite/blob/cc370d6cfef4a9d82761cc70fcb3bbeb0f91ab94/modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/IndexStorageImpl.java#L108 so I'll update this to do the same

Reply:
Imagine a table MyTable with 100 columns, and at some point you see the error:
`Encoded index name is too long for index: MyTable_189bbbb0-0c5f-3fb7-bba9-ad9285f193d1`.

How do you know which column  caused the issue?

Wouldn't you prefer something like:
`Encoded index name is too long for index MyTable_Column23_column34 `

We could also add in the message the name of the table and the column used to create the index as additional information.

Reply:
You also didn't add the charset StandardCharsets.UTF_8: 
 `byte[] idxNameBytes = idxName.getBytes(StandardCharsets.UTF_8);`

This means that if OGM runs on a machine with a different charset the validation might not be correct.